website/docs/reference/node-selection/defer.md

```diff
@@ -17,6 +17,24 @@ Defer requires that a manifest from a previous dbt invocation be passed to the `
 
 ### Usage
 
+<VersionBlock firstVersion="0.21">
+
+```shell
+$ dbt run --select [...] --defer --state path/to/artifacts
+$ dbt test --select [...] --defer --state path/to/artifacts
+```
+
+</VersionBlock>
+
+<VersionBlock firstVersion="0.20" lastVersion="0.20">
```

Comment:
does this imply that firstVersion is a `>=` and lastVersion is a `<=`? There could be behavioural changes in patches as well, are we choosing to not support that? If so then I'd advocate for the headers being 0.21.x, 1.0.x, etc

Reply:
Yes that's correct, although this is a early prototype so far to test how it will work. The `lastVersion` prop will be optional, so that value should only be set if a feature no longer exists but we still want to show it for an older version in the docs. This will prevent us from having to update all lastVersion props throughout the docs each time we add a new version to the site.

You make a good point regarding the patches. If a feature was released on version 0.20.3 for example, and we only have v0.20.0 and v1.0.2 in the versions dropdown, the user would have to select v1.0.2 to view that section. I think we should discuss this with @runleonarun to start narrowing down specifics on what versions we will have on the site, whether we should support patches, etc.

Reply:
I believe when @jtcohen6 and I talked about this, we decided to document major versions only. The idea being that minor versions would be bug fixes only that should not normally affect docs. I think 1.x is the furthest we'd need to slice the docs, right Jeremy?

Reply:
@runleonarun Right on, with just the minor clarification that `1.x` = "minor" and `1.x.y` = "patch".

To that end:

> If a feature was released on version 0.20.3 for example

There are two points here:
- We will not be releasing any new features in patch versions (such as v0.20.3), only minor versions (v0.21.0)
- We will not be releasing any new versions pre-v1.0

Reply:
Thanks for the details @jtcohen6, helpful to know what is expected from minor and patch releases. From both and editor and development standpoint, excluding patches from the versioning on the site will reduce the complexities of having to manage the versions overall.

It would be easier to maintain and add new blocks of versioned content, while also simplifying the logic quite a bit to check if the feature is available for the current selected version.

Reply:
@JKarlavige Here are some versioning clarifications that I discussed with Jeremy this week. I also captured them in our [notes for the week](https://www.notion.so/dbtlabs/March-8th-2022-26f5c53fe8864843974e3767a4bac31b).
    * We will not be surfacing patch version number or x. So, use 1.1 and not 1.1.x. We will message this in the Core versioning policy, which I captured in a follow-up task https://github.com/dbt-labs/docs.getdbt.com/issues/1217
    * Versioning in the picker will include `.20`,`.21`, `1.0` and higher. BUT we will remove `.20` and `.21` on June 30th, and communicate this in a banner that says they will no longer be supported after 6/30 and in the core versioning policy page. Also captured in a follow-up task https://github.com/dbt-labs/docs.getdbt.com/issues/1217
   
Let me know if there's anything else we can clarify!